backend/monitor/supervisor.go

```diff
@@ -174,27 +189,28 @@ func (m *EtcdSupervisor) getMonitor(ctx context.Context, key string) (*monitor,
 // watchMon takes a monitor key and watches for etcd ops. If a DELETE event
 // is witnessed, it calls the provided HandleFailure func. If a PUT event is
 // witnessed, the watcher is stopped.
-func watchMon(ctx context.Context, cli *clientv3.Client, mon *monitor, failureHandler func(), shutdownHandler func()) {
+func watchMon(ctx context.Context, cli *clientv3.Client, mon *monitor, failureHandler func(context.Context, int64), shutdownHandler func()) {
 	ctx, cancel := context.WithCancel(ctx)
 	responseChan := cli.Watch(ctx, mon.key)
 	leasesMu.Lock()
 	leasesToRevoke[mon.leaseID] = struct{}{}
 	leasesMu.Unlock()
 	go func() {
 		defer cancel()
+		ctx, cancelHandler := context.WithCancel(context.Background())
 		for {
 			select {
 			case wresp := <-responseChan:
 				for _, ev := range wresp.Events {
 					if ev.Type == mvccpb.DELETE {
-						failureHandler()
-						return
+						go failureHandler(ctx, mon.ttl)
```

Comment:
What does `mon.ttl` represent?
For keepalives, is that the keepalive interval?
For other objects, like checks, is that their timeout?

Reply:
`mon.ttl` is the time to live for both checks and keepalives.

Reply:
I was confused so I've re-read the docs for what timeout and ttl mean for Sensu. Now I think it makes sense to me. Thanks!